### DISCUSSION TITLE
Trying out the ZUGFeRDRenderer

### DISCUSSION THREAD

**Discussion by kadattack** [stephanstapel/ZUGFeRD-csharp#439] [Q&A]
I'm trying to understand what exactly the ZUGFeRDRenderer is doing so I ran code 

```
using ZUGFeRD_Test;
using s2industries.ZUGFeRDRenderer;

var invoice = new InvoiceProvider().CreateInvoice();
InvoiceDescriptorHtmlRenderer.render(invoice);
```

and get error

> Unhandled exception. System.ArgumentNullException: Value cannot be null. (Parameter 'stream')
>    at System.IO.StreamReader..ctor(Stream stream, Encoding encoding, Boolean detectEncodingFromByteOrderMarks, Int32 bufferSize, Boolean leaveOpen)
>    at s2industries.ZUGFeRDRenderer.InvoiceDescriptorHtmlRenderer.render(InvoiceDescriptor desc) in /home/newdev/Documents/Tehnolink/C#_Code/ZUGFeRD_Editor/ZUGFeRD_Editor/ZUGFeRD_Editor/ZUGFeRD-csharp/ZUGFeRDRenderer/InvoiceDescriptorHtmlRenderer.cs:line 44
>    at Program.<Main>$(String[] args) in /home/newdev/Documents/Tehnolink/C#_Code/ZUGFeRD_Editor/ZUGFeRD_Editor/ZUGFeRD_Editor/Program.cs:line 34

The line in question is https://github.com/stephanstapel/ZUGFeRD-csharp/blob/7e41377b0bc3492f43fd748bfe439290abf61c33/ZUGFeRDRenderer/InvoiceDescriptorHtmlRenderer.cs#L44

I assume this is happening because the `test.cshtml` is an empty file. 

My question is, what is `test.cshtml` actually suppose to look like and why was it left empty?

**Comment by stephanstapel** [comment]
Hi @kadattack ,
this was more a test project that I never put much work in.
Imho, it wouldn't be very difficult to turn this into a full fletched ZUGFeRD renderer. Let me add some elements to test.cshtml so that this can be easily understood. And if you like, I'd be more than happy if you could put some work into this.

**Comment by stephanstapel** [comment]
Look here:
https://github.com/stephanstapel/ZUGFeRD-csharp/commit/686ab11a4be3270ad2c140bfe2019a539a5a4d06

Imho this is a nice starting point. Surprisingly it is very easy to use Razor templates in a console application.
Just haven't found out yet if and how ViewComponents are possible, that'd be great for outputting party objects

**Comment by stephanstapel** [comment]
Hi @kadattack ,
did you have time to look into this?
